airflow/ui/src/dagsList.tsx

```diff
@@ -84,23 +89,100 @@ const columns: ColumnDef<DAG>[] = [
 
 type TableProps<TData> = {
   data: TData[];
+  total: number | undefined;
   columns: ColumnDef<TData>[];
   renderSubComponent: (props: { row: Row<TData> }) => React.ReactElement;
   getRowCanExpand: (row: Row<TData>) => boolean;
+  pagination: PaginationState;
+  setPagination: OnChangeFn<PaginationState>;
 };
 
+type PaginatorProps<TData> = {
+  table: TanStackTable<TData>;
+};
+
+function TablePaginator({ table }: PaginatorProps<DAG>) {
```

Comment:
Nitpick. Just for consistency's sake, let's make this an arrow function `TablePaginator = () => {`